Library/Homebrew/cleanup.rb

```diff
@@ -343,32 +343,32 @@ def cleanup_lockfiles(*lockfiles)
     end
 
     def cleanup_portable_ruby
-      system_ruby_version =
-        Utils.popen_read("/usr/bin/ruby", "-e", "puts RUBY_VERSION")
-             .chomp
-      use_system_ruby = (
-        Gem::Version.new(system_ruby_version) >= Gem::Version.new(RUBY_VERSION)
-      ) && !Homebrew::EnvConfig.force_vendor_ruby?
-      vendor_path = HOMEBREW_LIBRARY/"Homebrew/vendor"
-      portable_ruby_version_file = vendor_path/"portable-ruby-version"
-      portable_ruby_version = if portable_ruby_version_file.exist?
-        portable_ruby_version_file.read
-                                  .chomp
+      rubies = [which("ruby"), which("ruby", ENV["HOMEBREW_PATH"])]
+      system_ruby = Pathname.new("/usr/bin/ruby")
+      rubies << system_ruby if system_ruby.exist?
+
+      use_system_ruby = rubies.uniq.any? do |ruby|
+        system_ruby_version = Utils.safe_popen_read(ruby, "-e", "puts RUBY_VERSION").chomp
```

Comment:
Running this on my machine with `/usr/bin/time` takes 0.09s-0.10s. Running `/usr/bin/time ruby --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e "abort if Gem::Version.new(RUBY_VERSION.to_s.dup).to_s.split('.').first(2) != Gem::Version.new('2.6').to_s.split('.').first(2)"` takes 0.14s. I wonder if it's worth making that a shared `.rb` script file somewhere that can be run by both this code and `ruby.sh` below (considering both cases are having to shell out to Ruby). 

It's also worth noting this is only run when you manually run `brew cleanup` so it doesn't matter if it's a little slower if it ensure the code is completely in sync.

Reply:
Are you thinking about a Ruby script that would replace `test_ruby` bash function and that we would `require` and use in the `cleanup.rb`?

Reply:
@maxim-belkin Have pushed what I was thinking in 12d2e7ea4. Check it out and let me know what you think and, hopefully, it'll unblock you here.